test2code/src/main/kotlin/com/epam/drill/test2code/classparsing/Ast.kt

```diff
@@ -44,13 +47,30 @@ open class ProbeCounter : ClassProbesVisitor() {
 }
 
 class ClassProbeCounter(val name: String) : ProbeCounter() {
-    val astClass = newAstClass(name, ArrayList())
+    val astClass = AstEntity(
+        path = getPackageName(name),
+        name = getShortClassName(name),
+        methods = ArrayList(),
+        annotations = mutableMapOf()
+    )
 
     override fun visitMethod(
         access: Int, name: String?, desc: String?, signature: String?, exceptions: Array<out String>?
     ): MethodProbesVisitor {
         return MethodProbeCounter(astClass.methods as MutableList)
     }
+
+    override fun visitAnnotation(desc: String?, visible: Boolean): AnnotationVisitor? {
+        val annotationValues = mutableListOf<String>()
+
+        val annotationVisitor = object : AnnotationVisitor(api) {
+            override fun visit(name: String?, value: Any?) {
+                annotationValues.add(value.toString())
+            }
+        }
+        astClass.annotations[Type.getType(desc).className] = annotationValues
```

Comment:
create private field to accumulate data during visiting + add fun getAstClass to call after we done (to avoid mutable  types in AstEntity fields)

Reply:
done